sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -318,6 +318,112 @@ public virtual async Task<Response<LogsBatchQueryResultCollection>> QueryBatchAs
             }
         }
 
+        /// <summary>
+        /// Returns all the Azure Monitor logs matching the given query for an Azure resource.
+        /// <example snippet="Snippet:QueryResource">
+        /// <code language="csharp">
+        /// string workspaceId = &quot;&lt;workspace_id&gt;&quot;;
+        ///
+        /// var client = new LogsQueryClient(new DefaultAzureCredential());
+        ///
+        /// var results = await client.QueryResourceAsync(TestEnvironment.StorageAccountId,
+        ///     &quot;search *&quot;,
+        ///     new QueryTimeRange(TimeSpan.FromDays(5)));
+        ///
+        /// var resultTable = results.Value.Table;
+        ///
+        /// foreach (LogsTableRow rows in resultTable.Rows)
+        /// {
+        ///     foreach (var row in rows)
+        ///     {
+        ///         Console.WriteLine(row);
+        ///     }
+        /// }
+        ///
+        /// foreach (LogsTableColumn columns in resultTable.Columns)
+        /// {
+        ///     Console.WriteLine(&quot;Name: &quot; + columns.Name + &quot; Type: &quot; + columns.Type);
+        /// }
+        /// </code>
+        /// </example>
+        /// </summary>
+        /// <param name="resourceId"> The resourceId where the query should be executed. </param>
+        /// <param name="query"> The Kusto query to fetch the logs. </param>
+        /// <param name="timeRange"> The time period for which the logs should be looked up. </param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The logs matching the query.</returns>
+        public virtual Response<LogsQueryResult> QueryResource(string resourceId, string query, QueryTimeRange timeRange, CancellationToken cancellationToken = default)
```

Comment:
Should this be using the `ResourceIdentifier` type instead of string? /cc @m-nash

Reply:
We can add an overload method that takes in a ResourceIdentifier resourceId? @srnagar @tg-msft @m-nash

Reply:
Yes, we should use `ResourceIdentifier` - and if we add `"format": "arm-id"` in the swagger it would generate it automatically.